packages/distribution/addon/components/cd-inquiry-dialog.js

```diff
@@ -62,4 +72,47 @@ export default class CdInquiryDialogComponent extends Component {
 
     return response;
   }
+
+  @dropTask
+  *createInquiry(e) {
+    e.preventDefault();
+
+    try {
+      // get create inquiry work item to complete
+      const createId = decodeId(
+        this.distribution.controls.value?.create.edges[0].node.id
+      );
+
+      // create new inquiries
+      yield this.apollo.mutate({
+        mutation: createInquiryMutation,
+        variables: {
+          id: createId,
+          context: JSON.stringify({
+            addressed_groups: [String(this.args.to)],
+          }),
+        },
+      });
+
+      // refetch navigation and controls data
+      yield this.distribution.refetch();
+      // refetch dialog data
+      yield getObservable(this._inquiries.value).refetch();
+
+      this.router.transitionTo(
+        "inquiry.detail.index",
+        {
+          from: this.args.from,
+          to: this.args.to,
+        },
+        decodeId(this.inquiries[0].id)
+      );
+    } catch (e) {
+      this.notification.danger(
+        this.intl.t("caluma.distribution.new.error", {
+          count: 1,
+        })
+      );
+    }
```

Comment:
The code is very similar to the submit task of the cd-inquiry-new-form. Can we somehow centralize the create inquiry functionality and reuse it?

Reply:
Fair - done!